inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp

```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
+}
+
+TEST_P(OVInferRequestIODynamicTensorTest, compareOutputValuesDynamicShapes) {
+    ov::runtime::Tensor inTensor;
+    inTensor = ov::test::utils::create_and_fill_tensor(element::f32, {1, 4, 20, 20}, 10, 1);
+
+    OV_ASSERT_NO_THROW(req.set_tensor(tensor_name, inTensor));
+    OV_ASSERT_NO_THROW(req.infer());
+
+    ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+    ASSERT_EQ(inTensor.get_shape(), outTensor.get_shape());
+    ov::test::utils::compare(inTensor, outTensor, std::numeric_limits<float>::max(), std::numeric_limits<float>::max());
+}
+
+TEST_P(OVInferRequestIODynamicTensorTest, setTensorDynamicShapes) {
+    createModel({1, {1, 5}, {10, 20}, -1});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    for (int j = 0; j < 3; j++) {
+        ov::runtime::Tensor tensor(ngraph::element::Type_t::f32, {1, 4, 20, 20});
+        if (j == 2) tensor.set_shape({1, 2, 20, 20});
+
+        OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+        OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor));
+        OV_ASSERT_NO_THROW(req.infer());
+
+        ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+        ASSERT_EQ(tensor.get_shape(), outTensor.get_shape());
+        for (int i = 0; i < tensor.get_size(); i++) {
+            if (tensor.data<float>()[i] >= 0)
+                ASSERT_EQ(tensor.data<float>()[i], outTensor.data<float>()[i]);
+            else
+                ASSERT_EQ(0, outTensor.data<float>()[i]);
+        }
+    }
+}
+
+TEST_P(OVInferRequestIODynamicTensorTest, specialCase1) {
+    for (int j = 0; j < 5; j++) {
+        ov::runtime::Tensor actualTensor = req.get_tensor(tensor_name);
+        OV_ASSERT_NO_THROW(actualTensor.set_shape({1, 4, 20, 20}));
+
+        OV_ASSERT_NO_THROW(req.infer());
+
+        ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+        ASSERT_EQ(actualTensor.get_shape(), outTensor.get_shape());
+        for (int i = 0; i < actualTensor.get_size(); i++) {
+            if (actualTensor.data<float>()[i] >= 0)
+                ASSERT_EQ(actualTensor.data<float>()[i], outTensor.data<float>()[i]);
+            else
+                ASSERT_EQ(0, outTensor.data<float>()[i]);
+        }
+    }
+}
+
+TEST_P(OVInferRequestIODynamicTensorTest, specialCase2) {
+    createModel({-1, -1, -1, -1, -1, -1});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    float data[51200];
+    auto el = ngraph::element::Type_t::f32;
+
+    auto byteStrides = [](const ov::Strides& strides, const ov::element::Type& type) -> ov::Strides {
+        ov::Strides byte_strides(strides.size());
+        for (size_t i = 0; i < strides.size(); ++i)
+            byte_strides[i] = strides[i] * type.size();
+        return byte_strides;
+    };
+
+    ie::BlockingDesc blockingDesc1{{1, 4, 10, 10, 8, 16}, {0, 1, 2, 3, 4, 5}};
+    ov::runtime::Tensor tensor1(el, blockingDesc1.getBlockDims(), data, byteStrides(blockingDesc1.getStrides(), el));
```

Comment:
why do we need to pass byte strides explicitly? is it supposed that we test ROI?